src/services/statistic/index.js

```diff
@@ -7,129 +7,192 @@ const userModel = require('../user/model');
+			model: homeworkModel.submissionModel.find(schoolIdFilter),
+		},
+		{
+			name: 'classes',
+			promise: groupModel.classModel.countDocuments(schoolIdFilter),
+			model: groupModel.classModel.find(schoolIdFilter),
+		},
+		{
+			name: 'courses',
+			promise: groupModel.courseModel.countDocuments(schoolIdFilter),
+			model: groupModel.courseModel.find(schoolIdFilter),
+		},
+		{
+			name: 'teams',
+			promise: teamsModel.teamsModel.countDocuments(schoolIdFilter),
+			model: teamsModel.teamsModel.find(schoolIdFilter),
+		},
+		{
+			name: 'teachers',
+			promise: userModel.userModel.countDocuments(
+				schoolId
+					? { roles: roleTeacher, schoolId }
+					: { roles: roleTeacher },
+			),
+			model: userModel.userModel.find(
+				schoolId
+					? { roles: roleTeacher, schoolId }
+					: { roles: roleTeacher },
+			),
+		},
+		{
+			name: 'students',
+			promise: userModel.userModel.countDocuments(
+				schoolId
+					? { roles: roleStudent, schoolId }
+					: { roles: roleStudent },
+			),
+			model: userModel.userModel.find(
+				schoolId
+					? { roles: roleStudent, schoolId }
+					: { roles: roleStudent },
+			),
+		},
+	];
+
+	const promisesWithoutSchoolIds = [
+		{
+			name: 'schools',
+			promise: schoolModel.schoolModel.countDocuments(),
+			model: schoolModel.schoolModel.find(),
+		},
+		{
+			name: 'accounts',
+			promise: accountModel.countDocuments(),
+			model: accountModel.find(),
+		},
+		{
+			name: 'teams',
+			promise: teamsModel.teamsModel.countDocuments(),
+			model: teamsModel.teamsModel.find(),
+		},
+		{
+			name: 'lessons',
+			promise: lessonModel.countDocuments(),
+			model: lessonModel.find(),
+		},
+		{
+			name: 'files_directories',
+			promise: FileModel.countDocuments(),
+			model: FileModel.find(),
+		},
+		{
+			name: 'files_sizes',
+			promise: FileModel.aggregate([
+				{
+					$bucketAuto: {
+						groupBy: '$size',
+						buckets: 5,
+					},
 				},
-			},
-		]),
-		model: FileModel.find(),
-	},
-	{
-		name: 'files/types',
-		promise: FileModel.aggregate([
-			{
-				$group: {
-					_id: '$type',
-					total_files_per_type: { $sum: 1 },
+			]),
+			model: FileModel.find(),
+		},
+		{
+			name: 'files_types',
+			promise: FileModel.aggregate([
+				{
+					$group: {
+						_id: '$type',
+						total_files_per_type: { $sum: 1 },
+					},
 				},
-			},
-		]),
-		model: FileModel.find(),
-	},
-];
+			]),
+			model: FileModel.find(),
+		},
+	];
+
+	if (!schoolId) {
+		promises = [...promises, ...promisesWithoutSchoolIds];
+	}
 
-const fetchStatistics = () => {
+	return promises;
+};
+
+const fetchStatistics = (schoolId) => {
 	const statistics = {};
 
-	return Promise.all(
-		promises.map((p) => p.promise.exec().then((res) => {
-			statistics[p.name] = res;
-			return res;
-		})),
-	).then(() => statistics);
+	return Promise.all(getPromises(schoolId).map((p) => p.promise.exec().then((res) => {
+		statistics[p.name] = res;
+		return res;
+	}))).then((_) => statistics);
```

Comment:
Codacy found an issue: ['_' is already declared in the upper scope.](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-server/pullRequest?prid=6045807)